app/services/trainees/find_duplicates.rb

```diff
@@ -36,7 +36,7 @@ def matching_recruitment_cycle_year?(trainee)
     end
 
     def matching_qualification_type?(trainee)
-      trainee.training_route == course["route"]
+      course && trainee.training_route == course["route"]
```

Comment:
```suggestion
      course.present? && trainee.training_route == course["route"]
```

`matching_qualification_type` is that still the right method name?